mapbox_gl_web/lib/src/mapbox_web_gl_platform.dart

```diff
@@ -1050,4 +1050,9 @@ class MapboxWebGlPlatform extends MapboxGlPlatform
   void forceResizeWebMap() {
     _map.resize();
   }
+
+  @override
+  Future<bool> sourceExists(String sourceId) async {
+    return _map.hasImage(sourceId);
```

Comment:
`getSource` with a try catch should be the right option - as there is no direct option to query it